src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.DefaultItems.targets

```diff
@@ -22,6 +22,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     <EnableDefaultNoneItems Condition=" '$(EnableDefaultNoneItems)' == '' ">true</EnableDefaultNoneItems>
   </PropertyGroup>
 
+  <ItemGroup>
+    <None Remove="@(Compile)" Condition=" '$(EnableDefaultCompileItems)' != 'true' And '$(EnableDefaultNoneItems)' == 'true' " />
```

Comment:
Oh, I see from your description why this doesn't work.  Then the real question is:

Can you move all the other "Removes" to this location so they actually do something?  Especially this one:
`<None Remove="**/*$(DefaultLanguageSourceExtension)" Condition=" '$(EnableDefaultCompileItems)' == 'true' "/>`
since it is pretty much a dupe of what you have here.